driver/mounter_test.go

```diff
@@ -0,0 +1,110 @@
+package driver
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+)
+
+type testAttachmentValidator struct {
+	readFileFunc     func(name string) ([]byte, error)
+	evalSymlinksFunc func(path string) (string, error)
+}
+
+func (av *testAttachmentValidator) readFile(name string) ([]byte, error) {
+	return av.readFileFunc(name)
+}
+
+func (av *testAttachmentValidator) evalSymlinks(path string) (string, error) {
+	return av.evalSymlinksFunc(path)
+}
+
+func Test_mounter_IsAttached(t *testing.T) {
+	t.Run("could not evaluate the symbolic link", func(t *testing.T) {
```

Comment:
When I asked to inline `args` in a previous comment, I should have probably been clearer in saying that the anonymous `struct` slice is okay to keep in order to avoid code repetitions (which we currently have a bit in this case).

Would suggest something like this:

```go
tests := []test{
  name string
  source string
  av AttachmentValidator
  errorMsg string
  wantResult bool
}{
  {
    [...]
  }
}

for _, tt := range tests {
  t.Run(tt.name, func(t *testing.T) {
    // Set up fake mounter.
    gotResult, err := m.IsAttached(tt.source)
    if tt.errorMesg != "" {
      assert.ErrorContains(t, err, tt.errorMsg)
    } else {
      assert.Equal(t, gotResult, tt.wantResult)
    }
  })
}
```

This style of testing is a pattern we use very often.

The assertions come from the [gotest.tools library](https://pkg.go.dev/gotest.tools/v3) which I recently discovered and personally like very much. I am on a mission to introduce it wherever I can. :)

Reply:
Refactored the tests with the above.